### PR TITLE
framework/st_things : Add internal API st_things_stop.

### PR DESCRIPTION
**Pull request by Hyunjun85**
It will be able to support the restarting stack.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [ff0292edb0f7a20464011d3ba95400e455842db8] - Code Rule Check Result:
framework/src/st_things/st_things.c:280: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/st_things.c:460: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [ff0292edb0f7a20464011d3ba95400e455842db8] - Code Rule Check (C++) OK.

File: framework/include/st_things/st_things.h [hunk 1]
```diff
@@ -205,6 +205,8 @@ int st_things_register_reset_cb(st_things_reset_confirm_cb confirm_cb, st_things
  */
 int st_things_reset(void);
 
+int st_things_stop(void);
```
Comment (JeongJunSik) [inline framework/include/st_things/st_things.h]: please, remove this api,

File: framework/src/st_things/st_things.c [hunk 2]
```diff
@@ -449,6 +455,16 @@ int st_things_reset(void)
 	return ST_THINGS_ERROR_NONE;
 }
 
+int st_things_stop(void)
```
Comment (JeongJunSik) [inline framework/src/st_things/st_things.c]: Please remove this.

**Pull request closed without merging**